docs/design.md

```diff
@@ -91,97 +107,204 @@ This pattern is inspired by [sickle_ui](https://github.com/UmbraLuminosa/sickle_
-A limitation of this approach is that calling code cannot extend the spawn call with additional components or children.
-If you know about a better pattern, please let us know!
+A custom command is an elegant way to then indirectly call this function from anywhere in your code.
+
+A limitation of this approach is that calling code cannot extend the spawn call with additional components or children,
+as custom commands don't return `Entity` or `EntityCommands`. This kind of usage will be possible in future Bevy versions.
+
+## Interaction Callback
+
+### Pattern
+
+When adding a component that can have an interaction, such as a button that can be pressed,
+register a [one-shot system](https://bevyengine.org/news/bevy-0-12/#one-shot-systems) to handle the interaction.
+Add the resulting `SystemId` through a newtype component to your entity:
+
+```rust
+#[derive(Component, Debug, Reflect, Deref, DerefMut)]
+#[reflect(Component, from_reflect = false)]
+pub struct OnPress(#[reflect(ignore)] pub SystemId);
+
+
+fn spawn_button(mut commands: Commands) {
+    let pay_money = commands.register_one_shot_system(pay_money);
+    commands.button("Pay up!", pay_money);
+}
+
+
+// See the `Widgets` pattern for context
+impl<T: Spawn> Widgets for T {
+    fn button(&mut self, text: impl Into<String>, on_press: SystemId) -> EntityCommands {
+        self.spawn((
+            Name::new("Button"),
+            ButtonBundle {
+                // ...
+                default()
+            },
+            OnPress(on_press),
+        ));
+    }
+}
+```
+
+Note that the ugly reflect attributes are due to
+[`SystemId` not implementing `Reflect`](https://github.com/bevyengine/bevy/issues/14496).
+
+Also despawn the one-shot system when the interactive entity is despawned to prevent memory leaks:
+
+```rust
+pub(super) fn plugin(app: &mut App) {
+    app.observe(despawn_one_shot_system);
+}
+
+/// Remove the one-shot system entity when the [`OnPress`] component is removed.
+/// This is necessary as otherwise, the system would still exist after the button
+/// is removed, causing a memory leak.
+fn despawn_one_shot_system(
+    trigger: Trigger<OnRemove, OnPress>,
+    mut commands: Commands,
+    on_press_query: Query<&OnPress>,
+) {
+    let on_press = on_press_query.get(trigger.entity()).unwrap();
+    let one_shot_system_entity = on_press.entity();
+    commands.entity(one_shot_system_entity).despawn_recursive();
+}
+
+```
+
+### Reasoning
+
+This pattern is inspired by [bevy_mod_picking](https://github.com/aevyrie/bevy_mod_picking).
+By adding the system handling the interaction as a component to the entity,
+the code running on interactions does not need to differentiate about the entity it is running on.
+
+In simpler words, you don't need to check if the entity is the play button, the exit button,
+or something entire different, you just run the system that was added to the entity.
```

Comment:
I'm not sure about this section. It's framing the strength of this pattern as avoiding a drawback of another pattern that we haven't described. And we probably shouldn't describe it either, because that would be confusing.

Should just say something about how you get fully custom behavior for each button for free.

Reply:
That's fair.

Reply:
[`b78f813` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/b78f81344e315b331837da157cce176b66b8943e)